### PR TITLE
remove process.exit from tasks

### PR DESCRIPTION
**Pull request by bishalpd**

### CODE REVIEW

File: Tasks/AzureResourceGroupDeployment/main.ts [hunk 1]
```diff
@@ -26,13 +26,9 @@ function run(): Promise<void> {
...
-}
-catch (err) {
-    tl.setResult(tl.TaskResult.Failed, tl.loc("TaskNotFound", err));
-    process.exit();
-}
+var taskManifestPath = path.join(__dirname, "task.json");
+tl.debug("Setting resource path to " + taskManifestPath);
+tl.setResourcePath(taskManifestPath);
 
 run().then((result) =>
```
Comment (ds-ms) [inline Tasks/AzureResourceGroupDeployment/main.ts]: Isn't it better to move the run statement inside try?

Comment (bishalpd) [inline Tasks/AzureResourceGroupDeployment/main.ts]: In this case run already has a internal try-catch, hence not required.

Comment (ds-ms) [inline Tasks/AzureResourceGroupDeployment/main.ts]: Then we'd miss the "TaskNotFound" exception explicitly.

Comment (bishalpd) [inline Tasks/AzureResourceGroupDeployment/main.ts]: TaskNotFound should not needed to be caught at runtime. It should always work. At the best, it should be a debug.assert, but currently we do not have debug.assert in task lib. I will forward you Bryan's mail where he talked about this.

**Pull request merged**